### PR TITLE
RATIS-1158. Use the same proto for StateMachineExceptionProto, AlreadyClosedExceptionProto and DataStreamExceptionProto.

### PR DESCRIPTION
**Pull request by szetszwo**
See https://issues.apache.org/jira/browse/RATIS-1158

### CODE REVIEW

**Review by runzhiwang — APPROVED** [review]
LGTM, waiting CI passed.

**Comment by runzhiwang** [comment]
reopen pr to trigger ci

**Review by amaliujia — APPROVED** [review]
+1

File: ratis-proto/src/main/proto/Raft.proto [hunk 1]
```diff
@@ -321,22 +321,11 @@ message NotReplicatedExceptionProto {
...
-
-message DataStreamExceptionProto {
-  string exceptionClassName = 1;
-  string errorMsg = 2;
-  bytes stacktrace = 3;
+message ThrowableProto {
+  string className = 1;
+  string errorMessage = 2;
+  bytes stackTrace = 3;
+  bytes cause = 4;
```
Comment (amaliujia) [inline ratis-proto/src/main/proto/Raft.proto:328]: This is a good simplification 👍

**Comment by runzhiwang** [comment]
@szetszwo Thanks the patch. I have merged it.

**Pull request merged**